be/src/exec/pipeline/result_sink_operator.cpp

```diff
@@ -145,7 +145,9 @@ Status ResultSinkOperator::push_chunk(RuntimeState* state, const ChunkPtr& chunk
 
 Status ResultSinkOperatorFactory::prepare(RuntimeState* state) {
     RETURN_IF_ERROR(OperatorFactory::prepare(state));
-    RETURN_IF_ERROR(state->exec_env()->result_mgr()->create_sender(state->fragment_instance_id(), 1024, &_sender));
+    auto dop = state->query_options().pipeline_dop;
+    RETURN_IF_ERROR(state->exec_env()->result_mgr()->create_sender(state->fragment_instance_id(),
+                                                                   std::min(dop << 1, 1024), &_sender));
```

Comment:
Why change this?

Reply:
previous limit by row num, now by mem usage and block num.

Reply:
here set the limited block num